src/m365/planner/commands/task/task-remove.ts

```diff
@@ -0,0 +1,256 @@
+import { Group, PlannerBucket, PlannerTask } from '@microsoft/microsoft-graph-types';
+import { AxiosRequestConfig } from 'axios';
+import { Cli, Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { validation } from '../../../../utils';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+import { planner } from '../../../../utils/planner';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id?: string;
+  title?: string;
+  bucketId?: string;
+  bucketName?: string;
+  planId?: string;
+  planName?: string;
+  ownerGroupId?: string;
+  ownerGroupName?: string;
+  confirm?: boolean;
+}
+
+class PlannerTaskRemoveCommand extends GraphCommand {
+  public get name(): string {
+    return commands.TASK_REMOVE;
+  }
+  public get description(): string {
+    return 'Removes the Microsoft Planner task from a plan';
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    const removeTask: () => void = (): void => {
+      this
+        .getTask(args.options)
+        .then(task => {
+          const requestOptions: AxiosRequestConfig = {
+            url: `${this.resource}/v1.0/planner/tasks/${task.id}`,
+            headers: {
+              accept: 'application/json;odata.metadata=none',
+              'if-match': (task as any)['@odata.etag']
+            },
+            responseType: 'json'
+          }; 
+
+          return request.delete(requestOptions);
+        })
+        .then(_ => cb(), (err: any) => this.handleRejectedODataJsonPromise(err, logger, cb));
+    };
+    if (args.options.confirm) {
+      removeTask();
+    }
+    else {
+      Cli.prompt({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to remove the task ${args.options.id || args.options.title}?`
+      }, (result: { continue: boolean }): void => {
+        if (!result.continue) {
+          cb();
+        }
+        else {
+          removeTask();
+        }
+      });
+    }   
+  }
+
+  private getTask(options: Options): Promise<PlannerTask> {
+    const { id, title } = options;
+
+    if(id) {
+      const requestOptions: AxiosRequestConfig = {
+        url: `${this.resource}/v1.0/planner/tasks/${id}`,
+        headers: {
+          accept: 'application/json'
+        },
+        responseType: 'json'
+      };
+      
+      return request.get<PlannerTask>(requestOptions);
+    }
+
+    return this
+      .getBucketId(options)
+      .then(bucketId => {
+        const requestOptions: AxiosRequestConfig = {
+          url: `${this.resource}/v1.0/planner/buckets/${bucketId}/tasks`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
```

Comment:
The value `application/json;odata.metadate=none` causes the odata metadata, like the etag, to not get retrieved. While you need it further on. when specifying the `if-match` header. Let's change this accept header here to `application/json`.